python/cudf/cudf/_lib/io/utils.pyx

```diff
@@ -67,7 +67,10 @@ cdef source_info make_source_info(list src) except*:
 
 # Converts the Python sink input to libcudf++ IO sink_info.
 cdef sink_info make_sink_info(src, unique_ptr[data_sink] * sink) except*:
```

Comment:
This is unrelated to this PR, but it's fishy that we are accepting a `unique_ptr*` here. Does `sink_info` below expect to take ownership of the `unique_ptr`? If so, `make_sink_info` should also take ownership of it, i.e., accept a `unique_ptr` rather than a `unique_ptr *`.

Reply:
Looks like doing so is giving an error:

```bash
cudf/_lib/csv.cpp: In function ‘PyObject* __pyx_f_4cudf_4_lib_3csv_write_csv(__pyx_obj_4cudf_4_lib_5table_Table*, int, __pyx_opt_args_4cudf_4_lib_3csv_write_csv*)’:
cudf/_lib/csv.cpp:7946:102: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = cudf::io::data_sink; _Dp = std::default_delete<cudf::io::data_sink>]’
   __pyx_t_10 = __pyx_f_4cudf_4_lib_2io_5utils_make_sink_info(__pyx_v_path_or_buf, __pyx_v_data_sink_c); if (unlikely(PyErr_Occurred())) __PYX_ERR(0, 427, __pyx_L1_error)
                                                                                                      ^
In file included from /usr/include/c++/7/memory:80:0,
                 from cudf/_lib/csv.cpp:714:
/usr/include/c++/7/bits/unique_ptr.h:383:7: note: declared here
       unique_ptr(const unique_ptr&) = delete;
       ^~~~~~~~~~
In file included from cudf/_lib/csv.cpp:738:0:
/nvme/0/pgali/envs/cudfdev/include/python3.7m/datetime.h: At global scope:
/nvme/0/pgali/envs/cudfdev/include/python3.7m/datetime.h:204:25: warning: ‘PyDateTimeAPI’ defined but not used [-Wunused-variable]
 static PyDateTime_CAPI *PyDateTimeAPI = NULL;
                         ^~~~~~~~~~~~~
error: command 'gcc' failed with exit status 1
```

Reply:
Yeah that's because the `unique_ptr` must be passed to a function by moving it. This makes the intent clear that ownership is being passed to the function. For example:

```cython
unique_ptr[Foo] ptr = make_unique[Foo](...)

func(ptr) # passing unique_ptr by value is not valid
func(move(ptr)) # passing a moved unique_ptr, i.e., by rvalue-reference, is valid
```

Reply:
Had an offline discussion with @shwina regarding correct way to pass the unique_ptr to `make_sink_info` and made changes.